### ISSUE TITLE
sc2: CNN-based policy over 2D feature layer observations

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#103]
All current policies operate on a flat 13/21-dim vector. PySC2 provides rich 64×64 feature layers — `player_relative`, `unit_type`, `selected`, `height_map`, `unit_hit_points`, etc. — that encode spatial structure a convolutional policy can exploit. A CNN policy is essential for higher-level ladder play where spatial reasoning (army positioning, expansion locations, choke points) matters.

### New obs format when spatial layers are enabled

```python
# Dict observation space
{
    "flat":    Box(shape=(obs_dim,), dtype=float32),    # existing 13/21-dim vector
    "spatial": Box(shape=(C, 64, 64), dtype=float32),  # C selected screen layers
}
```

Config in `games/sc2/config/training_params.yaml`:
```yaml
screen_layers: [player_relative, selected]   # layer names from PySC2 feature_screen
minimap_layers: []                           # optional minimap channels
```

### Extraction in `SC2Client._timestep_to_obs_info()`

```python
if self._screen_layers:
    spatial = np.stack([
        obs["feature_screen"][LAYER_INDEX[name]] / LAYER_SCALE[name]
        for name in self._screen_layers
    ]).astype(np.float32)   # shape (C, 64, 64)
    info["spatial_obs"] = spatial
```

## CNN policy architecture

```
spatial (C, 64, 64)
    │
Conv2d(C → 32, 3×3, relu)
Conv2d(32 → 64, 3×3, relu)
AdaptiveAvgPool2d(4×4)
Flatten → (1024,)
    │
Concat with flat obs (obs_dim,)
    │
FC(1024 + obs_dim → 256, relu)
    │
  ┌───┴────┐
fn_head   spatial_head
  (6,)       (9,)
```

Weights evolved by CMA-ES (same as `LSTMEvolutionPolicy` — evolutionary search over flattened numpy weight array, no backprop required).

## Training loop integration

`train_rl()` in `framework/training.py` detects dict obs space and routes to `CNNPolicy`:
```python
if isinstance(env.observation_space, gym.spaces.Dict):
    policy = CNNPolicy(obs_spec, screen_layers, ...)
else:
    policy = make_policy(policy_type, obs_spec, ...)
```

## Tasks

- [ ] Add `screen_layers` / `minimap_layers` to `SC2ObsSpec` and parse from training config
- [ ] Extend `SC2Client._timestep_to_obs_info()` to stack selected feature layers into `info["spatial_obs"]`
- [ ] Change `SC2Env.observation_space` to `gym.spaces.Dict` when spatial obs is enabled; update `reset()` and `step()` to return dict obs
- [ ] Implement `SC2CNNPolicy` in `games/sc2/cnn_policy.py`: conv layers → flatten → concat flat → two FC heads (fn_idx + spatial); evolved via CMA-ES over flat weight vector
- [ ] Update `train_rl()` to detect dict obs space and instantiate `SC2CNNPolicy`
- [ ] Sensible defaults in `games/sc2/config/training_params.yaml`:
  ```yaml
  screen_layers: [player_relative, selected]
  ```
- [ ] Tests: CNN forward pass with correct input shapes `(C, 64, 64)`; flat+spatial concat dimension; spatial obs extracted from mock PySC2 timestep; dict obs space returned from `env.reset()`

## Relevant files

- `games/sc2/env.py` — observation space definition, `reset()` / `step()`
- `games/sc2/client.py` — `_timestep_to_obs_info()`, feature layer extraction
- `games/sc2/obs_spec.py` — `SC2ObsSpec`, `get_spec()`
- `framework/training.py` — `train_rl()`, policy factory
- New: `games/sc2/cnn_policy.py`
